### ISSUE TITLE
ananse enhancer: ValueError: Cannot take a larger sample than population when 'replace=False'

### ISSUE THREAD

**Issue by andrewbcaldwell** [vanheeringen-lab/ANANSE#33]
Hi,
I am attempting to run ananse enhancer using the hg38 genome (installed with genomepy) for ATAC-seq data. The peaks were called with Genrich and the bam file is sorted, and the index exists. Here is my command:
```
ananse enhancer -g hg38 -t ATAC \
>                   -b celltype2.bam \
>                   -p celltype2.Genrich.peaks.narrowPeak \
>                   -o celltype2.ananse.enhancer.bed
```
and this is the error I get:
```
Could not find/load indexes.
Traceback (most recent call last):
  File "/home/andrewbcaldwell/miniconda3/envs/ananse/bin/ananse", line 364, in <module>
    args.func(args)
  File "/home/andrewbcaldwell/miniconda3/envs/ananse/lib/python3.8/site-packages/ananse/commands/enhancer.py", line 43, in enhancer
    b.run_enhancer(
  File "/home/andrewbcaldwell/miniconda3/envs/ananse/lib/python3.8/site-packages/ananse/enhancer.py", line 234, in run_enhancer
    quantile_bed = self.quantileNormalize(epeak, bed_cov, bed_output)
  File "/home/andrewbcaldwell/miniconda3/envs/ananse/lib/python3.8/site-packages/ananse/enhancer.py", line 213, in quantileNormalize
    rank = pd.read_csv(self.peak_rank, header=None).sample(n = enahcer_number, random_state = 1).sort_values(0)[0].tolist()
  File "/home/andrewbcaldwell/miniconda3/envs/ananse/lib/python3.8/site-packages/pandas/core/generic.py", line 4995, in sample
    locs = rs.choice(axis_length, size=n, replace=replace, p=weights)
  File "mtrand.pyx", line 959, in numpy.random.mtrand.RandomState.choice
ValueError: Cannot take a larger sample than population when 'replace=False'`
```
I've been able to run the whole pipeline (ananse enhancer through influence) for 3/4 of my cell types, but for some reason I run into this issue for 1 cell type. Processing for bam files and peak calling parameters were the same for all 4 cell types. Do you have any idea what may be causing this error? Thanks!

**Comment by Maarten-vd-Sande** [comment]
Hi @andrewbcaldwell, thanks for the issue.

The current approach of ananse is to fit the pileup of the reads under the peaks onto a h3k27ac distribution we found when comparing many different cell types / tissues. This guarantees that everything comes from the same (biological "correct") distribution. However this also means we have to sample from that distribution. In your case, you have more peaks than we have in our reference dataset.

We recently had a discussion in our group and we agreed that this feels _off_. We are currently in the process of changing this to use the distribution of your sample(s). This would resolve these kinds of problems.

For now there are three options:

1 wait until we release the changed version (this is going to take a while, better not!)
2 call peaks more stringent, so you have less peaks
3 go into the source code and patch it. Go to `File "/home/andrewbcaldwell/miniconda3/envs/ananse/lib/python3.8/site-packages/ananse/enhancer.py", line 213` and add `replace=True` to the `.sample` call

For options 2 and 3, it is best to re-do peak calling or ananse for all samples again respectively

**Comment by Maarten-vd-Sande** [comment]
I believe this has been fixed since release 2.0, so I am closing this. Feel free to reopen if it still occurs.
